nautobot/extras/tests/test_customfields.py

```diff
@@ -29,6 +29,25 @@ def setUp(self):
         Site.objects.create(name="Site B", slug="site-b", status=active_status)
         Site.objects.create(name="Site C", slug="site-c", status=active_status)
 
+    def test_custom_field_dict_population(self):
+        """Test that custom_field_data is properly populated when no data is passed in."""
+        name = "Custom Field"
+        custom_field = CustomField.objects.create(
+            # 2.0 TODO: #824 remove name field
+            name=name,
+            slug="custom_field",
+            type=CustomFieldTypeChoices.TYPE_TEXT,
+        )
+        custom_field.validated_save()
+        custom_field.content_types.set([ContentType.objects.get_for_model(Site)])
+
+        site = Site.objects.create(name="Test", status=Status.objects.get(name="Active"))
```

Comment:
Please avoid using `Site` in new code where possible since it makes more work for us in 2.0. :-) I might suggest `Provider` as an alternate simple model to use instead.

Reply:
Makes sense.